lib/restore.js

```diff
@@ -91,44 +92,53 @@ Restore.prototype.handle = function(request, response) {
 Restore.prototype.dispatch = function(request, response) {
   var method = request.method.toUpperCase(),
       uri    = url.parse(request.url, true),
-      match  = null;
+      match  = null,
+      startBaseURL = new RegExp('^\/?' + this._baseURL + '\/?');
 
   request.secure = this.isSecureRequest(request);
 
+  if (!uri.pathname.match(startBaseURL)) {
+    response.writeHead(302, { 'Location': this._baseURL})
+    return response.end()
+  }
+
+  // remove baseURL and eventually a final '/' before real path
+  uri.pathname = uri.pathname.replace(startBaseURL, '')
```

Comment:
You can replace with `'/'` instead of `''` and avoid all those changes to the regexes further below.